vars/rpmlintMockResults.groovy

```diff
@@ -47,50 +23,34 @@ void call(String config, Boolean allow_errors=false, Boolean skip_rpmlint=false,
     if (fileExists('utils/rpms/Makefile')) {
         chdir = 'utils/rpms/'
     }
-    String output = sh(label: 'RPM Lint built RPMs',
-                       script: 'cd ' + chdir + '\n' +
+    int result = sh(label: 'RPM Lint built RPMs',
+                    script: 'cd ' + chdir + '\n' +
                              /* groovylint-disable-next-line GStringExpressionWithinString */
                              '''name=$(make ''' + make_args + ''' show_NAME)
                                 if [ -f "$name".rpmlintrc ]; then
                                     rpmlint_args=(-r "$name".rpmlintrc)
                                 fi
                                 rpmlint --ignore-unused-rpmlintrc "${rpmlint_args[@]}" ''' +
                                '$(ls /var/lib/mock/' + config + '/result/*.rpm | ' +
-                               'grep -v -e -debuginfo -e debugsource) || exit 0',
-                       returnStdout: true).trim()
-
-    int result = sh(label: 'Analyze rpmlint output',
-                    script: "read e w b < <(echo \"${output}\" | " +
-                          """sed -nEe '\$s/.*([0-9]+) errors, ([0-9]+) warnings, ([0-9]+) badness;.*/\\1 \\2 \\3/p')
-                               if [ "\$e" -gt 0 ]; then
-                                   exit 2
-                               elif [ "\$w" -gt 0 ]; then
-                                   exit 1
-                               fi
-                               exit 0""",
+                               'grep -v -e -debuginfo -e debugsource)',
                     returnStatus: true)
 
     catchError(stageResult: 'UNSTABLE', buildResult: 'SUCCESS') {
         if (result == 2 && allow_errors) {
```

Comment:
Previously `result` was either 0, 1, or 2.  Now that it should be the return status from rpmlint, we probably just want to check to see if `result` is greater than 0 - which means errors detected.  If we want to catch warnings as well, one option would be to convert all warnings into errors with a `setBadness('warning', 1000)` entry in the daos.rpmlintrc file.

Reply:
Updated